core/src/test/java/com/graphhopper/routing/util/parsers/BikeTagParserTest.java

```diff
@@ -677,4 +678,23 @@ public void temporalAccess() {
         assertTrue(accessEnc.getBool(false, edgeId, access));
     }
 
+    @Test
+    public void testPedestrian() {
+        ReaderWay way = new ReaderWay(1);
+        way.setTag("highway", "pedestrian");
+        assertPriorityAndSpeed(SLIGHT_AVOID, 4, way);
+
+        way.setTag("bicycle", "yes");
+        way.setTag("highway", "pedestrian");
+        assertPriorityAndSpeed(PREFER, 12, way);
+
+        way.setTag("bicycle", "yes");
+        way.setTag("surface", "asphalt");
+        way.setTag("highway", "pedestrian");
+        assertPriorityAndSpeed(PREFER, 12, way);
+
+        way.setTag("cycleway:right", "track");
+        way.setTag("highway", "pedestrian");
+        assertPriorityAndSpeed(PREFER, 18, way);
+    }
```

Comment:
I find following easier to read:
```
    @Test
    public void testPedestrian() {
        ReaderWay way = new ReaderWay(1);
        way.setTag("highway", "pedestrian");
        assertPriorityAndSpeed(SLIGHT_AVOID, 4, way);
        way.setTag("bicycle", "yes");
        assertPriorityAndSpeed(PREFER, 12, way);
        way.setTag("surface", "asphalt");
        assertPriorityAndSpeed(PREFER, 12, way);

        way.clearTags();
        way.setTag("highway", "pedestrian");
        way.setTag("cycleway:right", "track");
        way.setTag("bicycle", "yes");
        assertPriorityAndSpeed(PREFER, 18, way);
    }
```
The last check currently requires the `bicycle=yes` tag to be present, otherwise the priority would result in SLIGHT_AVOID. It would be nice if we could classify this also as PREFER also without `bicycle=yes` as this tag is kind of redundant for this edge case in combination with `cycleway:right`